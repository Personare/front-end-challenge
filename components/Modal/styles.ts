import styled,{ keyframes } from "styled-components"
import media from 'styled-media-query'

const scaleIn = keyframes`
  0% {
    transform: scale(0);
  }
  100% {
    transform: scale(1);
  }
`

export const Modal = styled.div`
  background-color: rgba(255, 255, 255, 0.35);
  backdrop-filter: blur(5px);
  padding: .7rem;
  flex-direction: column;  
  border-radius: .5rem;
  display: flex;
  align-items: center;
  position: relative;
  -webkit-animation: ${scaleIn} 700ms cubic-bezier(0.250, 0.460, 0.450, 0.940);
  animation: ${scaleIn} 700ms cubic-bezier(0.250, 0.460, 0.450, 0.940);
  `
export const ModalBackDrop = styled.div`
  background-color: rgba($color: #000000c2, $alpha: 0.35);
  height: 100%;
  position: fixed;
  top: 0;
  left: 0;
  display: flex;
  justify-content: center;
  align-items: center; 
`
export const ModalContainer = styled.div`
  width: 60%;
  ${media.lessThan('medium')`
    max-height: 100%;
    width: 90%;
  `}
`

export const ModalContent = styled.div`
  border-radius: .5rem;
  padding: 0.2rem; 
  backdrop-filter: blur(100px);
`
