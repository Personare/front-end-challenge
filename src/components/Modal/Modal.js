import React, { Component } from 'react'
import styled from 'styled-components'
import {
  func, node, arrayOf, oneOfType, string
} from 'prop-types'

class Modal extends Component {
  constructor(props) {
    super(props)
    this.handleKeyUp = this.handleKeyUp.bind(this)
  }

  componentDidMount() {
    window.addEventListener('keyup', this.handleKeyUp, false)
  }

  componentWillUnmount() {
    window.removeEventListener('keyup', this.handleKeyUp, false)
  }

  handleKeyUp(e) {
    const { onClose } = this.props
    if (e.keyCode === 27) {
      e.preventDefault()
      onClose()
      window.removeEventListener('keyup', this.handleKeyUp, false)
    }
  }

  render() {
    const { onClose, children, title } = this.props

    return (
      <Wrapper>
        <ModalOverlay onClick={onClose} />
        <ModalContainer>
          <ModalContent>
            <ModalHeader>
              {title}
              <CloseIcon onClick={onClose} />
            </ModalHeader>
            <ChildrenWrapper>{children}</ChildrenWrapper>
          </ModalContent>
        </ModalContainer>
      </Wrapper>
    )
  }
}

const Wrapper = styled.div`
  visibility: ${props => props.visibilityStatus};
`

const ModalOverlay = styled.div`
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  height: 100%;
  width: 100%;
  z-index: 1000;
  background: rgba(0, 0, 0, 0.76);
`

const ModalContainer = styled.div.attrs({
  'data-modal': 'modal-content'
})`
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 1000;
  overflow: auto;
  text-align: center;
  padding: 4px;
  cursor: pointer;
`

const ModalContent = styled.div.attrs({
  'data-modal': 'modal-dialog'
})`
  position: relative;
  outline: 0;
  width: auto;
  display: inline-block;
  vertical-align: middle;
  box-sizing: border-box;
  cursor: default;
  border-radius: 3px;
  width: auto;
  margin: 30px auto;
  background-color: #ffffff;

  @media (max-width: 575px) {
    min-width: 100%;
    height: 100%;
    overflow: auto;
    margin: 0;
  }
`

const ChildrenWrapper = styled.div`
  display: flex;
  justify-content: center;
`

const ModalHeader = styled.div`
  margin: 10px 0;
  font-size: 18px;
  font-weight: 600;
`

const CloseIcon = styled.img.attrs({
  src: './assets/close_icon.svg'
})`
  width: 30px;
  height: 30px;
  position: absolute;
  right: 0;
  top: 0;
  margin: 5px;
`

Modal.propTypes = {
  onClose: func.isRequired,
  children: oneOfType([arrayOf(node), node]).isRequired,
  title: string
}

Modal.defaultProps = {
  title: ''
}
export default Modal
