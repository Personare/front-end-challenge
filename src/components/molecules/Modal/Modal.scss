.Modal {
    position: fixed;
    left: 0;
    top: 0;
    z-index: 1000;
    width: 100%;
    height: 100vh;
    background-color: rgba(0,0,0,0.2);
    display: flex;
    align-items: center;
    justify-content: center;
    font-size: 4rem;
    color: #000;

    &__content {
        position: relative;
        background: #fff;
        padding: 3.6rem;
        width: 100%;
        max-width: 100rem;
        height: 100%;
        text-align: center;
        overflow-y: scroll;

        h1 {
            font-size: 2rem;
            margin-bottom: 1.4rem;
        }

        p {
            color: #333;
            font-size: 1.8rem;
        }

        @media (min-width: 768px) {
            height: auto;
        }
    }

    &__close {
        position: absolute;
        width: 3rem;
        height: 3rem;
        top: 1rem;
        right: 1rem;
        border: 0;
        background: 0;
        cursor: pointer;

        svg {
            stroke: #000;
        }
    }
}

