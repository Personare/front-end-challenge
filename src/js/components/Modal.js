import React, { Component } from 'react';

class Modal extends Component {
    constructor(props) {
        super(props);
    }

    render() {
        let { card, modalState, onClick } = this.props;

        return (
            <div className={'Modal is-' + modalState }>
                <span className='Modal__close' onClick={onClick}>X</span>

                {card && (
                    <div className='Modal__wrapper'>
                        <div className='CardPicture'>
                            <img className='CardPicture__image' src={card.imgUrl} />
                        </div>

                        <div className='Modal__content'>
                            <h1 className='primary-title'>{card.name}</h1>
                            <p className='body-text'>
                                Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s,
                                when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries,
                                but also the leap into electronic typesetting, remaining essentially unchanged.
                            </p>
                        </div>
                    </div>
                )}
            </div>
        );
    }
};

export default Modal;
