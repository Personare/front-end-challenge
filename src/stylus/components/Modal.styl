.Modal
    display block
    position fixed
    top 50%
    left 50%
    transform translate(-50%, -50%)
    background #ffffff
    width calc(100% - 40px)
    max-width calc(1200px - 40px)
    border-radius 10px
    padding 20px
    min-height 60vh
    z-index 2
    transition transform 0.3s cubic-bezier(0.175, 0.885, 0.32, 1.275), opacity 0.3s cubic-bezier(0.175, 0.885, 0.32, 1.275)

    &.is-hide
        transform translate(-50%, 50%)

.Modal__close
    display block
    position absolute
    cursor pointer
    top -40px
    right 0
    width 30px
    height 30px
    background #f3f3f3
    text-align center
    line-height 30px
    color #2d2d2d
    border-radius 10px
    font-family 'Roboto', sans-serif
    font-size 2em
    font-weight 700

    &:hover
        background #fff

.Modal__wrapper
    display flex
    text-align center
    flex-wrap wrap

    +below(breakpoints.phone)
        flex-direction column

.Modal__content
    display inline-block
    flex 1
    padding-left 20px

    +below(breakpoints.phone)
        width 100%
        padding-left 0

