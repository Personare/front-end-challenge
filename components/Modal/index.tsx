import * as S from './styles'

type ModalProps={
    onBackdropClick: () => void;
    children: React.ReactNode;
}

const Modal = ({onBackdropClick, children}: ModalProps) =>(  
  <S.ModalBackDrop onClick={onBackdropClick}>
      <S.ModalContainer>
        <S.Modal>
          <S.ModalContent>
            {children}
          </S.ModalContent>
        </S.Modal>
    </S.ModalContainer>
  </S.ModalBackDrop>
);


export default Modal
