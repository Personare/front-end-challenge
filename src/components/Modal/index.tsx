import { ICard } from "../../config/interfaces";
import Card from "../Card";
import { Content, Text, Overlay, Name, CloseButton } from "./styles";

interface IModal {
  card: ICard;
  onClose: () => void;
}

const Modal: React.FC<IModal> = ({ card, onClose }) => {
  const handleContentClick = (e: any) => {
    e.stopPropagation();
  };

  return (
    <Overlay onClick={onClose} data-cy="modal-overlay">
      <Content onClick={handleContentClick}>
        <Name>{card.name}</Name>
        <Card card={card} />
        <Text>
          The standard chunk of Lorem Ipsum used since the 1500s is reproduced
          below for those interested. Sections 1.10.32 and 1.10.33 from "de
          Finibus Bonorum et Malorum" by Cicero.
        </Text>

        <CloseButton onClick={onClose}>&times;</CloseButton>
      </Content>
    </Overlay>
  );
};
export default Modal;
