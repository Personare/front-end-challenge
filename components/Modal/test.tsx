import { render } from "@testing-library/react"
import Modal from './'

describe('components/Card', () => {
  it('Should match snapshot', () => {
    const { container } = render(<Modal onBackdropClick={jest.fn} ><h1>n1</h1></Modal>)
    expect(container).toMatchSnapshot()
  })
  
})
