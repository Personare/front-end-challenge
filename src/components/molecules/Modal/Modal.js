import React from 'react';
import ReactDOM from 'react-dom';
import PropTypes from 'prop-types';
import './Modal.scss';

const Modal = ({ isOpen, name, image, close }) => {
    const setScrollLock = value => (document.querySelector('html').style.overflow = value);

    if (!isOpen) {
        setScrollLock('');
        return null;
    }

    setScrollLock('hidden');

    return ReactDOM.createPortal(
        <div className="Modal" onClick={close}>
            <div className="Modal__content" onClick={e => e.stopPropagation()}>
                <button className="Modal__close" onClick={close}>
                    <svg viewBox="0 0 40 40">
                        <path d="M 10,10 L 30,30 M 30,10 L 10,30" />
                    </svg>
                </button>
                <img src={image} alt={name} />
                <h1>{name}</h1>
                <p>
                    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam mauris felis, pulvinar cursus
                    pulvinar at, pretium vitae sem. Fusce dolor elit, bibendum sed mollis lobortis, ornare sit amet
                    libero. Proin congue, urna et semper pharetra, ex diam auctor metus, ac malesuada lectus sapien non
                    tortor. Maecenas metus augue, auctor at varius vel, facilisis non mi. Sed risus mi, convallis a
                    lobortis ac, cursus et tortor. Aenean in tempus nisl, a viverra turpis.
                </p>
            </div>
        </div>,
        document.body
    );
};

Modal.propTypes = {
    name: PropTypes.string,
    image: PropTypes.string
};

export default Modal;
