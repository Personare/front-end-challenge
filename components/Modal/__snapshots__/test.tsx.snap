// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Card Should match snapshot 1`] = `
.c2 {
  background-color: rgba(255,255,255,0.35);
  -webkit-backdrop-filter: blur(5px);
  backdrop-filter: blur(5px);
  padding: .7rem;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border-radius: .5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-animation: gGreRZ 700ms cubic-bezier(0.250,0.460,0.450,0.940);
  -webkit-animation: gGreRZ 700ms cubic-bezier(0.250,0.460,0.450,0.940);
  animation: gGreRZ 700ms cubic-bezier(0.250,0.460,0.450,0.940);
}

.c0 {
  background-color: rgba($color:#000000c2,$alpha:0.35);
  height: 100%;
  position: fixed;
  top: 0;
  left: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  width: 60%;
}

.c3 {
  border-radius: .5rem;
  padding: 0.2rem;
  -webkit-backdrop-filter: blur(100px);
  backdrop-filter: blur(100px);
}

@media (max-width:768px) {
  .c1 {
    max-height: 100%;
    width: 90%;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <h1>
            n1
          </h1>
        </div>
      </div>
    </div>
  </div>
</div>
`;
